HW8/Nudyk.V/password.py

```diff
@@ -0,0 +1,24 @@
+import re
+
+
+def password(user_input):
+    ''' Checks if the password is valid
+    '''
+
+    if  re.findall('[a-z]', user_input) == [] or\
+    re.findall('[A-Z]', user_input) == [] or\
+    re.findall('[0-9]', user_input) == [] or\
+    re.findall('[$,#,@]', user_input) == [] or\
+    len(user_input) < 6 or\
+    len(user_input) > 16:
+        return 'Bad password, pal'
+    else:
+        return 'Very good password, hacker'
+
+
+
+print(password('a'))
+print(password('aZ'))
+print(password('aZ1'))
+print(password('aZ1@'))
+print(password('aZ1@aZ1@'))
```

Comment:
another way

import re

password = input("Enter your password:\n")

if re.match("^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[$#@])(?=.{6,16}$).*", password):
    print("Nice pass! Psss, I won't tell anyone ;)")
else:
    print("Password is not valid. Check if you have:"
          """
    At least 1 letter between[a-z] and 1 letter between[A-Z].
    At least 1 number between[0-9].
    At least 1 character from [$#@].
    Minimum length 6 characters.
    Maximum length 16 characters.
          """
           )
####################

import re

def validate_password(input_password):
    
    flag = True
    
    while flag:  
        if len(input_password)<6 or len(input_password)>16:
            break
        elif not re.search("[a-z]",input_password):
            break
        elif not re.search("[0-9]",input_password):
            break
        elif not re.search("[A-Z]",input_password):
            break
        elif not re.search("[$#@]",input_password):
            break
        # Returns a match where the string contains a white space character
        elif re.search("\s",input_password):  
            break
        else:
            return f"{input_password} is valid password"
            flag = False
            break

    if flag:
        return f"{input_password} is invalid password"


print(validate_password("ER45jknkjbyhdfy"))
print(validate_password("ER45@hjkyutt TYTIU566"))
print(validate_password("DFG567#oijl89"))